facts/coatOfArms/coat_of_arms.py

```diff
@@ -0,0 +1,116 @@
+    from either P94 (coat of arms image) or P154 (logo image) properties.
+    
+    Args:
+        municipality_name (str): Name of the municipality to search for
+        
+    Returns:
+        Optional[str]: URL to coat of arms image, or None if not found 
+    """
+
+
+    wiki_ids = get_municipality_wikiId(municipality_name)
+
+    if not wiki_ids: 
+        return None
+   
+    
+    headers = {"User-Agent": "KlimatkollenFetcher/1.0 (contact: hej@klimatkollen.se)"}
+    coat_of_arms_url = None
+   
+    for id in wiki_ids:
+        url = f'https://www.wikidata.org/w/api.php?action=wbgetentities&ids={id}&props=claims&format=json'
+        res = requests.get(url, headers=headers) 
+ 
+        try: 
+            response = res.json()
+            claims = response["entities"][id]["claims"]
+
+            p94 = claims.get("P94")
+            p154 = claims.get("P154")
+            filename = None 
+
+            if p94:
+                print("P94 exists:", p94[0])
+
+                for statement in p94:
+                    snak = statement.get("mainsnak",{})
+                    if "datavalue" in snak:
+                        filename = snak["datavalue"]["value"].replace(" ", "_")
+                        break
+                
+                if filename and isinstance(filename, str):
+                    print(filename)
+                    url = f"https://commons.wikimedia.org/wiki/Special:Redirect/file/{quote(filename)}"
+                    res = requests.get(url, headers=headers, allow_redirects=True) 
+                    coat_of_arms_url = res.url
+
+            else:
+                print("No P94 exists, trying P154...")
+
+                if p154:
+                    for statement in p154:
+                        snak = statement.get("mainsnak",{})
+                        if "datavalue" in snak:
+                            filename = snak["datavalue"]["value"].replace(" ", "_")
+                            break
+                    
+                    if filename and isinstance(filename, str):
+                        url = f"https://commons.wikimedia.org/wiki/Special:Redirect/file/{quote(filename)}"
+                        res = requests.get(url, headers=headers, allow_redirects=True) 
+                        coat_of_arms_url = res.url
+
+                else:
+                    print("No P154 exists")
+
+        except ValueError:
+            print("Could not parse response to JSON")
+            return
+    print(coat_of_arms_url)
+    return coat_of_arms_url
+
+
+def get_municipality_wikiId(municipality_name):
+    url = "https://www.wikidata.org/w/api.php"
+
+    params = {
+        "action": "wbsearchentities",
+        "search": municipality_name,
+        "language": "sv",
+        "format": "json"
+    }
+
+    headers = {
+        "User-Agent": "KlimatkollenFetcher/1.0 (contact: hej@klimatkollen.se)"
+    }
+
+    res = requests.get(url, params=params, headers=headers)
```

Comment:
You could add a timeout here - to prevent program to hang, good custom:


```suggestion
    res = requests.get(url, params=params, headers=headers, timeout=30)
```